### PR TITLE
fix: Include year in response

### PR DESCRIPTION
**Pull request by LarsV123**
Includes the "year" field in responses when fetching Journal, Publisher and Series, which was previously omitted. Note that if the external API does not include this, the field value is taken from the request instead.

### CODE REVIEW

**Comment by LarsV123** [comment]
Note: Some noise in the PR due to differences in style between my IntelliJ configuration and existing code. Attempted to avoid this, but there does not seem to be a single, consistent rule set for code style for the project?

**Comment by LarsV123** [comment]
Ticket: https://sikt.atlassian.net/browse/NP-46722

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/journal/FetchJournalByIdentifierAndYearHandlerTest.java [hunk 1]
```diff
@@ -124,6 +122,22 @@ void shouldReturnCorrectDataWithSuccessWhenExists() throws IOException {
...
+    void shouldIncludeYearInResponse() throws IOException {
+        // Arrange
+        var year = TestUtils.randomYear();
+        var identifier = mockRegistry.randomJournal(year);
+        var input = constructRequest(String.valueOf(year), identifier);
+
+        // Act
+        handlerUnderTest.handleRequest(input, output, context);
+
+        // Assert
```
Comment (truhacevkir) [inline src/test/java/no/sikt/nva/pubchannels/handler/fetch/journal/FetchJournalByIdentifierAndYearHandlerTest.java:135]: Do we need these comments :)

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/series/FetchSeriesByIdentifierAndYearHandlerTest.java [hunk 2]
```diff
@@ -106,6 +105,23 @@ void shouldReturnCorrectDataWithSuccessWhenExists() throws IOException {
...
+        // Arrange
+        var year = randomYear();
+        var identifier = UUID.randomUUID().toString();
+        var input = constructRequest(String.valueOf(year), identifier, MediaType.ANY_TYPE);
+        mockSeriesFound(year, identifier);
+
+        // Act
+        handlerUnderTest.handleRequest(input, output, context);
+
+        // Assert
```
Comment (truhacevkir) [inline src/test/java/no/sikt/nva/pubchannels/handler/fetch/series/FetchSeriesByIdentifierAndYearHandlerTest.java:119]: Same here

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/publisher/FetchPublisherByIdentifierAndYearHandlerTest.java [hunk 3]
```diff
@@ -107,6 +106,23 @@ void shouldReturnCorrectDataWithSuccessWhenExists() throws IOException {
         assertThat(actualPublisher, is(equalTo(expectedPublisher)));
     }
 
+    @Test
+    void shouldIncludeYearInResponse() throws IOException {
+        // Arrange
```
Comment (truhacevkir) [inline src/test/java/no/sikt/nva/pubchannels/handler/fetch/publisher/FetchPublisherByIdentifierAndYearHandlerTest.java:111]: Same here

File: src/main/java/no/sikt/nva/pubchannels/handler/model/SeriesDto.java [hunk 4]
```diff
@@ -120,10 +124,21 @@ public String getDiscontinued() {
...
 
+    public String getYear() {
+        return year;
+    }
+
     @Override
     @JacocoGenerated
     public int hashCode() {
-        return Objects.hash(context, id, identifier, name, onlineIssn, printIssn, scientificValue, sameAs,
+        return Objects.hash(context,
```
Comment (truhacevkir) [inline src/main/java/no/sikt/nva/pubchannels/handler/model/SeriesDto.java:134]: Add year to hashCode() og equals()

**Review by truhacevkir — COMMENTED** [review]
Ser bra ut, få kommentarer

**Review by torbjokv — APPROVED** [review]
lgtm

**Review by monaullah — APPROVED** [review]
Nice work!

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/journal/FetchJournalByIdentifierAndYearHandlerTest.java (hunk 1 above)
Comment (LarsV123) [reply src/test/java/no/sikt/nva/pubchannels/handler/fetch/journal/FetchJournalByIdentifierAndYearHandlerTest.java]: Nope, removed.

File: src/main/java/no/sikt/nva/pubchannels/handler/model/SeriesDto.java (hunk 4 above)
Comment (LarsV123) [reply src/main/java/no/sikt/nva/pubchannels/handler/model/SeriesDto.java:134]: Thanks, fixed!

**Pull request merged**